src/main/java/com/bmc/truesight/meter/plugin/remedy/RemedyTicketsCollector.java

```diff
@@ -71,59 +73,77 @@ public void run() {
+                            }
 
-                        if (totalRecordsRead < nMatches.longValue() && (totalRecordsRead + chunkSize) > nMatches.longValue()) {
-                            //assuming the long value would be in int range always
-                        	chunkSize = (int) (nMatches.longValue() - totalRecordsRead);
-                        }else if(totalRecordsRead >= nMatches.longValue()){
-                        	readNext=false;
+                            if (eventList.size() > 0) {
+                                List<String> eventsList = new ArrayList<>();
+                                eventList.forEach(event -> {
+                                    Gson gson = new Gson();
+                                    String eventJson = gson.toJson(event, Object.class);
+                                    StringBuilder sendEventToTSI = new StringBuilder();
+                                    sendEventToTSI.append(Constants.REMEDY_PROXY_EVENT_JSON_START_STRING).append(eventJson).append(Constants.REMEDY_PROXY_EVENT_JSON_END_STRING);
+                                    // eventSinkAPI.emit(sendEventToTSI.toString());
+                                    eventsList.add(sendEventToTSI.toString());
+                                });
+                                int succ = eventSinkAPI.emit(eventsList);
+                                System.err.println(succ + " Events successfuly ingested out of total " + nMatches + " events in iteration " + iteration);
+                            } else {
+                                System.err.println(eventList.size() + " Events found for the interval, DateTime:" + template.getConfig().getStartDateTime() + " to DateTime:" + template.getConfig().getEndDateTime());
+                                eventSinkAPIstd.emit(Util.eventMeterTSI(Constants.REMEDY_PLUGIN_TITLE_MSG, Constants.REMEDY_IM_NO_DATA_AVAILABLE, Event.EventSeverity.INFO.toString()));
+                            }
+
+                            if (totalRecordsRead < nMatches.longValue()
+                                    && (totalRecordsRead + chunkSize) > nMatches.longValue()) {
+                                // assuming the long value would be in int range
+                                // always
+                                chunkSize = (int) (nMatches.longValue() - totalRecordsRead);
+                            } else if (totalRecordsRead >= nMatches.longValue()) {
+                                readNext = false;
+                            }
+                            iteration++;
+                            startFrom = totalRecordsRead;
+                        }//each chunk iteration
+                        //close the RPC connection 
+                        if (eventSinkAPI.closeRPCConnection()) {
```

Comment:
We may remove this.. anyways, you have handled closeconnection in finally block.. that is enough